raphtory/src/io/parquet_loaders.rs

```diff
@@ -0,0 +1,315 @@
+    let size = cols_to_check.len();
+
+    load_node_props_from_df(
+        &df,
+        size,
+        id,
+        const_properties,
+        shared_const_properties,
+        graph,
+    )
+    .map_err(|e| GraphError::LoadFailure(format!("Failed to load graph {e:?}")))?;
+
+    Ok(())
+}
+
+pub fn load_edge_props_from_parquet<
+    G: StaticGraphViewOps + InternalPropertyAdditionOps + InternalAdditionOps,
+>(
+    graph: &G,
+    parquet_file_path: &Path,
+    src: &str,
+    dst: &str,
+    const_properties: Option<Vec<&str>>,
+    shared_const_properties: Option<HashMap<String, Prop>>,
+    layer: Option<&str>,
+    layer_in_df: Option<bool>,
+) -> Result<(), GraphError> {
+    let mut cols_to_check = vec![src, dst];
+    if layer_in_df.unwrap_or(false) {
+        if let Some(ref layer) = layer {
+            cols_to_check.push(layer.as_ref());
+        }
+    }
+    cols_to_check.extend(const_properties.as_ref().unwrap_or(&Vec::new()));
+
+    let df = process_parquet_file_to_df(parquet_file_path, cols_to_check.clone())?;
+    df.check_cols_exist(&cols_to_check)?;
+    let size = cols_to_check.len();
+
+    load_edges_props_from_df(
+        &df,
+        size,
+        src,
+        dst,
+        const_properties,
+        shared_const_properties,
+        layer,
+        layer_in_df.unwrap_or(true),
+        graph,
+    )
+    .map_err(|e| GraphError::LoadFailure(format!("Failed to load graph {e:?}")))?;
+
+    Ok(())
+}
+
+pub fn load_edges_deletions_from_parquet<
+    G: StaticGraphViewOps + InternalPropertyAdditionOps + InternalAdditionOps + DeletionOps,
+>(
+    graph: &G,
+    parquet_file_path: &Path,
+    src: &str,
+    dst: &str,
+    time: &str,
+    layer: Option<&str>,
+    layer_in_df: Option<bool>,
+) -> Result<(), GraphError> {
+    let mut cols_to_check = vec![src, dst, time];
+    if layer_in_df.unwrap_or(true) {
+        if let Some(ref layer) = layer {
+            cols_to_check.push(layer.as_ref());
+        }
+    }
+
+    let df = process_parquet_file_to_df(parquet_file_path, cols_to_check.clone())?;
+    df.check_cols_exist(&cols_to_check)?;
+    let size = cols_to_check.len();
+
+    load_edges_deletions_from_df(
+        &df,
+        size,
+        src,
+        dst,
+        time,
+        layer,
+        layer_in_df.unwrap_or(true),
+        graph,
+    )
+    .map_err(|e| GraphError::LoadFailure(format!("Failed to load graph {e:?}")))?;
+
+    Ok(())
+}
+
+pub(crate) fn process_parquet_file_to_df(
+    parquet_file_path: &Path,
+    col_names: Vec<&str>,
+) -> Result<PretendDF, GraphError> {
+    let (names, arrays) = read_parquet_file(parquet_file_path, &col_names)?;
+
+    let names = names
+        .into_iter()
+        .filter(|x| col_names.contains(&x.as_str()))
+        .collect();
+    let arrays = arrays
+        .map_ok(|r| r.into_iter().map(|boxed| boxed.clone()).collect_vec())
+        .collect::<Result<Vec<_>, _>>()?;
+
+    Ok(PretendDF { names, arrays })
+}
```

Comment:
We should make this work when the `parquet_file_path` is a directory of parquet files. This is a pretty common format, e.g., if the data was generated by Spark.